stock_valuation_fifo_lot/models/stock_move.py

```diff
@@ -0,0 +1,122 @@
+            if not move._is_in():
+                continue
+            for ml in move._get_in_move_lines():
+                ml.qty_base = ml.product_uom_id._compute_quantity(
+                    ml.qty_done, ml.product_id.uom_id
+                )
+        return res
+
+    def _get_move_lots(self):
+        self.ensure_one()
+        correction_ml = self.env.context.get("correction_move_line")
+        return correction_ml.lot_id if correction_ml else self.lot_ids
+
+    def _prepare_common_svl_vals(self):
+        """Add lots/serials to the stock valuation layer."""
+        self.ensure_one()
+        res = super()._prepare_common_svl_vals()
+        lots = self._get_move_lots()
+        res.update({"lot_ids": [Command.set(lots.ids)]})
+        return res
+
+    def _create_out_svl(self, forced_quantity=None):
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            # Set the move as a context for processing in _run_fifo().
+            move = move.with_context(fifo_move=move)
+            layer = super(StockMove, move)._create_out_svl(
+                forced_quantity=forced_quantity
+            )
+            product = move.product_id
+            # To prevent unknown creation of negative inventory.
+            if (
+                product.cost_method == "fifo"
+                and product.tracking != "none"
+                and layer.remaining_qty < 0
+            ):
+                raise UserError(
+                    _("Negative inventory is not allowed for product %s.")
+                    % product.display_name
+                )
+            layers |= layer
+        return layers
+
+    def _create_in_svl(self, forced_quantity=None):
+        correction_ml = self.env.context.get("correction_move_line")
+        if forced_quantity and correction_ml:
+            correction_ml.qty_base += forced_quantity
+            return super()._create_in_svl(forced_quantity=forced_quantity)
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            layer = super(StockMove, move)._create_in_svl(
+                forced_quantity=forced_quantity
+            )
+            layers |= layer
+            # Calculate standard price (sorted by lot created date)
+            product = move.product_id
+            if product.cost_method != "fifo" or product.tracking == "none":
+                continue
+            for ml in layer.stock_move_id.move_line_ids:
+                ml.qty_base = ml.qty_done
+            # Change product standard price to the first available lot price.
+            product = product.with_context(sort_by="lot_create_date")
+            candidate = product._get_fifo_candidates(move.company_id)[:1]
+            if not candidate:
+                continue
+            product = product.with_company(move.company_id.id)
+            product = product.with_context(disable_auto_svl=True)
+            product.sudo().standard_price = candidate.unit_cost
```

Comment:
I'm actually not clear about the business rationale why we need this part.

@ps-tubtim @Saran440 Could you please explain what business needs you had to have this logic? In my view, product standard price doesn't mean much in FIFO (it doesn't affect the transaction except for the case where inventory is created with a new lot with no purchase order) and this logic can be removed.

Reply:
@yostashiro https://github.com/OCA/stock-logistics-workflow/blob/15.0/stock_valuation_fifo_lot/models/stock_move.py
As I check the same file in version 15, I don't see that part of code.

Reply:
@kittiu Sorry, we made it hard to track. It's this part: https://github.com/OCA/stock-logistics-workflow/blob/a87e986/stock_valuation_fifo_lot/models/stock_move.py#L47-L62

Reply:
@yostashiro
As far as I remember, this part will affect the case where you return items to lot in the warehouse.

**Example**

1. Receive goods (WH/IN) with lot001 10 unit, price/unit = $10 => Standard price = $10 (Price from lot001)
2. Receive goods (WH/IN) with lot002 5 unit, price/unit = $15 => Standard price = $10 (Price from lot001)
3. Issue goods (WH/OUT) 12 unit => Standard price = $15 (Price from lot002)
4. Return goods (WH/RET) lot001 3 unit => Standard price = $10 (Price from lot001)

This part of code will check when incoming goods to warehouse, the system will pull price from first lot always.

Reply:
@newtratip Thanks for the explanation. I understand your logic, but I don't fully understand what happens if we don't update the `standard_price` in this scenario. I assume that we don't care about the `standard_price` for FIFO products. Could you please explain the problem if we don't update the standard_price in this case?

Reply:
@kittiu @newtratip I removed this part as I don’t see any business need for it, and I don't think we need to be concerned about the standard price for FIFO products. Please let me know if I’m missing something regarding the business or operations.

Reply:
@AungKoKoLin1997 I believe the standard price is used in at least 2 places, even for FIFO products:

- Sales order line cost and margin
- Inventory adjustment / new quant

https://github.com/odoo/odoo/blob/bc7b2154cbaf49c33af4452f78300cb791e2e3db/addons/sale_margin/models/sale_order_line.py#L27

Reply:
@matt454357 Thanks for the pointer. I think this part is useful in cases where the first lot of a product is delivered in full and then some quantity is returned. In this situation, the standard price is updated to match the price of the first lot.

So, I think using the standard price of the first lot for the Sales Order line cost doesn't have an impact. Am I misunderstanding?

For inventory adjustments or new quants, adjustments for existing inventory will use the lot price of that inventory with the current design. When creating a new quant, which means a new lot, the standard price of the first or any other lot isn’t necessarily relevant, as I wouldn’t expect it to use the same standard price. So, user should check the standard price before they create new quant.